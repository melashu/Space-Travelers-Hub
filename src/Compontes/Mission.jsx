import React, { useEffect } from 'react';
import Table from 'react-bootstrap/Table';
import 'bootstrap/dist/css/bootstrap.min.css';
import { Badge, Button } from 'react-bootstrap';
import { useDispatch, useSelector } from 'react-redux';
import {
  getLoadingState,
  getMissionState,
  joinMission,
  leaveMission,
  missionThunk,
} from '../Store/reducers';

const Mission = () => {
  const missions = useSelector(getMissionState);
  const loadingStatus = useSelector(getLoadingState);
  const dispatch = useDispatch();
  useEffect(() => {
    if (missions.length === 0) dispatch(missionThunk());
  }, [dispatch, missions.length]);
  const rows = missions.map((row) => (
    <tr key={row.mission_id}>
      <td>{row.mission_name}</td>
      <td>{row.description}</td>
      <td className="">
        {' '}
        {row.reserved ? (
          <Badge bg="primary">Active Member </Badge>
        ) : (
          <Badge bg="secondary">NOT Member </Badge>
        )}
      </td>
      <td className="button-container status">
        {' '}
        {row.reserved ? (
          <Button
            size="lg"
            variant="outline-danger"
            onClick={() => {
              dispatch(leaveMission(row.mission_id));
            }}
          >
            Leave Missions
          </Button>
        ) : (
          <Button
            variant="outline-secondary"
            size="lg"
            onClick={() => {
              dispatch(joinMission(row.mission_id));
            }}
          >
            Joining Missions
          </Button>
        )}
      </td>
    </tr>
  ));
  return (
    <>
      {loadingStatus === 'idel' && (
        <Badge bg="primary" className="loadingStatus">
          Initializing....
        </Badge>
      )}
      {loadingStatus === 'pending' && (
        <Badge bg="info" className="loadingStatus">
          Pending....
        </Badge>
      )}
      {loadingStatus === 'rejected' && (
        <Badge bg="danger" className="loadingStatus">
          Something wents wrong
        </Badge>
      )}
      {loadingStatus === 'fulfilled' && (
        <Table striped hover size="md" responsive bordered>
          <thead>
            <tr>
              <th>Mission</th>
              <th>Description</th>
              <th>Status</th>
            </tr>
          </thead>
          <tbody>{rows}</tbody>
        </Table>
      )}
    </>
  );
};

export default Mission;
