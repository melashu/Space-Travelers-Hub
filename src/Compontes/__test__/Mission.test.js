import { render, screen } from "@testing-library/react";
import "@testing-library/jest-dom";
import Mission from "../Mission";
import { Provider } from "react-redux";
import renderer from "react-test-renderer";

const mission = () => {
  <Provider store={store}>
    <Mission />
  </Provider>;
};

describe("Should mission render correctly", () => {
  it("Test Mission Componenet", () => {
    const component = renderer.create(<mission />).toJSON();
    expect(component).toMatchSnapshot();
  });

  it("Test Mission Componenet", () => {
    const component = renderer.create(<mission />).toJSON();
    expect(component).toMatchInlineSnapshot(`<mission />`);
  });
});
