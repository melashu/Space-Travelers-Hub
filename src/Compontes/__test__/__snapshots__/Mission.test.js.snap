// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should mission render correctly Mission Componenet 1`] = `<mission />`;

exports[`Should mission render correctly Test Mission Componenet 1`] = `<mission />`;

exports[`Should mission render correctly Test Mission Componenet 2`] = `<mission />`;
